src/electionguard/discrete_log.py

```diff
@@ -0,0 +1,100 @@
+# pylint: disable=global-statement
+# support for computing discrete logs, with a cache so they're never recomputed
+
+import asyncio
+from typing import Dict, Tuple
+
+from electionguard.singleton import Singleton
+
+from .group import G, ElementModP, ONE_MOD_P, mult_p, int_to_p_unchecked
+
+DLOG_CACHE = Dict[ElementModP, int]
+DLOG_MAX = 1_000_000
```

Comment:
i set the default here to 1m but it could be something else.  it was 100m in dan's example

Reply:
This should do something to what happens in the primes. This could be set with environment variables. Default to this but be overridable via environment.